File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] arr, int target) {
        int[] ans = new int[2];
        int st = 0;
        int end = arr.length - 1;

        while(st <= end ) {
            int sum = arr[st] + arr[end];
            if(sum == target) {
                ans[0] = st + 1;
                ans[1] = end + 1; 
                return ans;
            } else if(sum > target) {
                end--;
            } else {
                st++;
            }
        }
        return ans;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        int st = 0;
        int end = n - 1;
        
        while (st < end) {
            int sum = nums[st] + nums[end];
            
            if (sum == target) return {st+1, end+1};
            
            if (sum > target) {
                end--;
            } else {
                st++;
            }
        }
        return {};
    }
};